src/app/main/main.controller.js

```diff
@@ -100,15 +100,29 @@ angular.module('transmartBaseUi')
      * @param node Dropped node from the study tree
      */
     $scope.onNodeDropEvent = function (event, info, node) {
-      // Makes the progress bar animated
-      $scope.cohortUpdating = true;
-      CohortSelectionService.nodes.push(node);
-      ChartService.addNodeToActiveCohortSelection(node).then(function () {
-        $scope.cohortUpdating = false;
-        ChartService.updateDimensions();
-      });
+      //if the dragged node is not at the leaf level
+      if(node.type !== 'CATEGORICAL_OPTION') {
+          if(CohortSelectionService.nodes.indexOf(node) == -1) {
+              $scope.addNodeToCohortSelection(node);
+          }
+      }
+      else {
+          //if the node's parent's chart has not been created, create it
+          if(CohortSelectionService.nodes.indexOf(node.parent) == -1) {
+              $scope.addNodeToCohortSelection(node.parent);
+          }
+      }
     };
```

Comment:
It might be nicer to have this logic inside the ChartService.addNodeToActiveCohortSelection instead inside this controller